### PR TITLE
Customer Invoice Test Cases

### PR DESCRIPTION
**Pull request by ashaneo**
This PR includes 2 test cases for;

- DOT-TC-677: Verify the Cut+Dry Pay in customer invoices

> Login to Distributor application
> Click on Customers from the left side bar
> Verify that user navigated to Customers section
> Verify that no error displayed in Customers section
> Click on search box and search (customer code : 16579) customer then click
> on that customer
> Verify that user navigated to the customer profile section
> Click on 'Invoices' tab
> Verify that user navigate to the invoices section
> Verify that Cut+Dry Pay status

- DOT-TC-678: Verify the Payment Terms in customer invoices

> Login to Distributor application
> Click on Customers from the left side bar
> Verify that user navigated to Customers section
> Verify that no error displayed in Customers section
> Click on search box and search (customer code : 16579) customer then click
> on that customer
> Verify that user navigated to the customer profile section
> Click on 'Invoices' tab
> Verify that user navigate to the invoices section
> Verify that Payment Terms status

### CODE REVIEW

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java [hunk 1]
```diff
@@ -349,8 +352,11 @@ public class CustomersPage extends LoginPage {
     By btn_statusSave = By.xpath("//button[contains(@class,'mr-2 my-2 btn btn-outline')]");
```
Comment (AhsanMubarak) [inline src/main/java/com/cutanddry/qa/pages/CustomersPage.java:352]: Can't we get any unique attribute except the class name. It's containing more than 1 class

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java [hunk 2]
```diff
@@ -349,8 +352,11 @@ public class CustomersPage extends LoginPage {
...
     By txt_status = By.xpath("//div[contains(text(),'Active')]");
     By ls_status = By.xpath("//div[contains(@class,'themed_select__menu-list css-11unzgr')]");
```
Comment (AhsanMubarak) [inline src/main/java/com/cutanddry/qa/pages/CustomersPage.java:354]: Can't we get any unique attribute except the class name. It's containing more than 1 class

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java [hunk 3]
```diff
@@ -1899,6 +1905,69 @@ public boolean isActiveStatusDisplayed(){
...
+            // Wait to see if the error text becomes visible
+            wait.until(ExpectedConditions.presenceOfElementLocated(txt_error));
+            return false; // "error" text is found
+        } catch (TimeoutException e) {
+            return true; // No "error" text is found within the timeout
+        }
+    }
+
+    public void clickOnFirstItemOfCreditRequests(){
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
```
Comment (AhsanMubarak) [inline src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1920]: Please create the wait method in the KeywordBase class and reuse it here. We don't need to define the time here and we can use default one

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java [hunk 4]
```diff
@@ -1899,6 +1905,69 @@ public boolean isActiveStatusDisplayed(){
...
+
+    public void clickonInvoice(){
+        distributorUI.click(btn_invoice);
+    }
+
+    public void verifyEnabledStatus() {
+
+        // Wait until at least one matching element is visible
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(enabledStatusLocator));
```
Comment (AhsanMubarak) [inline src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1933]: Please reuse the existing method from the KeywordBase class fro waiting

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java [hunk 5]
```diff
@@ -1899,6 +1905,69 @@ public boolean isActiveStatusDisplayed(){
...
+        } else {
+            System.out.println("No element with 'Enabled' text found.");
+        }
+    }
+
+    public void verifyPaymentTermStatus() {
 
+        // Wait until at least one matching element is visible
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(defaultTermStatusLocator));
```
Comment (AhsanMubarak) [inline src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1955]: Please reuse the existing method from the KeywordBase class fro waiting

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import org.testng.ITestResult;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class VerifythePaymentTermsincustomerInvoices extends TestBase
+{
+    static User user;
+    String timeRange = "All";
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:19]: Keep this data separate from testcases

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        Login.loginAsDistributor(user.getEmailOrMobile(), user.getPassword());
+        Dashboard.isUserNavigatedToDashboard();
+        softAssert.assertTrue(Dashboard.isUserNavigatedToDashboard(),"login error");
+        Dashboard.navigateToCustomers();
+        softAssert.assertTrue(Customer.isErrorTextNotDisplayed(),"Error Message Displayed");
+        Customer.searchCustomerByCode(Customer_Code);
+        Customer.clickOnFirstItemOfCreditRequests();
+        Customer.clickonInvoice();
+        Customer.verifyPaymentTermStatus();
+
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:40]: Please don't forget to close the assertion 
softAssert.assertAll();

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        initialization();
+        user = JsonUtil.readUserLogin();
+    }
+
+    @Test(groups = "DOT-TC-678")
+    public void VerifythePaymentTermsincustomerInvoices() throws InterruptedException {
+        SoftAssert softAssert = new SoftAssert();
+        Login.loginAsDistributor(user.getEmailOrMobile(), user.getPassword());
+        Dashboard.isUserNavigatedToDashboard();
+        softAssert.assertTrue(Dashboard.isUserNavigatedToDashboard(),"login error");
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:33]: Just a suggestion, Please update the error message as meaningful
ex: "The user is unable to land on the Dashboard page."

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        SoftAssert softAssert = new SoftAssert();
+        Login.loginAsDistributor(user.getEmailOrMobile(), user.getPassword());
+        Dashboard.isUserNavigatedToDashboard();
+        softAssert.assertTrue(Dashboard.isUserNavigatedToDashboard(),"login error");
+        Dashboard.navigateToCustomers();
+        softAssert.assertTrue(Customer.isErrorTextNotDisplayed(),"Error Message Displayed");
+        Customer.searchCustomerByCode(Customer_Code);
+        Customer.clickOnFirstItemOfCreditRequests();
+        Customer.clickonInvoice();
+        Customer.verifyEnabledStatus();
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:34]: Please don't forget to close the assertion
softAssert.assertAll();

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        initialization();
+        user = JsonUtil.readUserLogin();
+    }
+
+    @Test(groups = "DOT-TC-677")
+    public void VerifytheCutnandDryPayincustomerinvoices() throws InterruptedException {
+        SoftAssert softAssert = new SoftAssert();
+        Login.loginAsDistributor(user.getEmailOrMobile(), user.getPassword());
+        Dashboard.isUserNavigatedToDashboard();
+        softAssert.assertTrue(Dashboard.isUserNavigatedToDashboard(),"login error");
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:28]: Just a suggestion, Please update the error message as meaningful
ex: "The user is unable to land on the Dashboard page."

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+import com.cutanddry.qa.functions.Dashboard;
+import com.cutanddry.qa.functions.Login;
+import com.cutanddry.qa.utils.JsonUtil;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class verifytheCutnandDryPayincustomerinvoices extends TestBase {
+    static User user;
+    String Customer_Code = "16579";
```
Comment (AhsanMubarak) [inline src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:15]: Keep this data separate from testcases

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 1 above)
Comment (ashaneo) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:352]: Hi, I checked on this but this path is not written by me

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 2 above)
Comment (ashaneo) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:354]: Hi, I checked on this but this path is not written by me

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 3 above)
Comment (ashaneo) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1920]: Done

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 4 above)
Comment (ashaneo) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1933]: Done

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 5 above)
Comment (ashaneo) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:1955]: Done

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java (hunk 6 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:19]: Done, file created in data folder

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java (hunk 7 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:40]: done

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java (hunk 8 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/VerifythePaymentTermsincustomerInvoices.java:33]: Done

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java (hunk 10 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:28]: Changed

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java (hunk 11 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:15]: Done

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 1 above)
Comment (AhsanMubarak) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:352]: Okay, Noted

File: src/main/java/com/cutanddry/qa/pages/CustomersPage.java (hunk 2 above)
Comment (AhsanMubarak) [reply src/main/java/com/cutanddry/qa/pages/CustomersPage.java:354]: Noted

File: src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java (hunk 9 above)
Comment (ashaneo) [reply src/test/java/com/cutanddry/qa/tests/customer_invoice/verifytheCutnandDryPayincustomerinvoices.java:34]: Done

**Review by AhsanMubarak — APPROVED** [review]

**Pull request merged**